docs/public/specifications/tests/Address-TP/address02.md

```diff
@@ -1,46 +1,102 @@
+
+## Scope
+
+This test checks for the existence of PTR records for the corresponding reverse
+domains of the name servers IP address.
+
+## Inputs*
 
 The domain name to be tested.
 
-### Ordered description of steps to be taken to execute the test case
+## Summary
+
+Message Tag                   | Level    | Arguments | Message ID for message tag
+:---------------------------- |:---------|:----------|:--------------------------
+A02_PTR_PRESENT               | INFO     |           | PTR present for all nameservers
+A02_PTR_MISSING               | NOTICE   | ns_list   | PTR missing for "{ns_list}"
+
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine profile]. Also see the
+[Severity Level Definitions] document.
+
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [argument list].
+
 
-1. Obtain the glue address records  of the domain checked
-   using [Method4](../Methods.md)
+## Test procedure 
 
-2. Obtain the IP addresses of each name server of the domain checked
-   using [Method5](../Methods.md)
+1. Create the empty set: Name server name and IP address ("Name Server IP").
 
-3. For each IP address, a recursive PTR query must be performed.
+2. Obtain the glue address records of each name server for the domain from the
+   parent using the method [Get-Del-NS-Names-and-IPs] and add them to the 
+   *Name Server IP* set. 
 
-4. If any answer of the queries performed in step 3 contains an RCODE
-   other than NOERROR or if the answer does not include a PTR record,
-   this test case fails.
+3. Obtain the IP addresses of each name server for the domain using the method 
+   [Get-Zone-NS-Names-and-IPs] and add any non-duplicate results to 
+   *Name Server IP* set. 
 
-### Outcome(s)
+4. Create the following empty set: IP address ("PTR Missing")
 
-If the test case succeeds, its result is a list of addresses with corresponding
-hostnames which are the result of the PTR queries performed.
-The result could be represented as a hash table where the keys are the IP
-addresses and the values their corresponding hostnames.
+5. For each name server in *Name Server IP* do:
+   1. Make a recursive PTR query.
+   2. If the response fails to match the following criteria, add the IP address
+      to the *PTR Missing* set.
+        - RCODE must be NOERROR
+        - answer section must contain at least one PTR record
+  
+6. If the set *PTR Missing* is empty, then putput *[A02_PTR_PRESENT]*
 
-### Special procedural requirements
+7. Else, output *[A02_PTR_MISSING]* with a list of the IP addresses in the 
+   *PTR Missing* set.
+
+
+## Outcome(s)
+
+The outcome of the Test Case is "notice" if there is at least one message with
+the message with the severity level *[NOTICE]*.
+
+If there are no message or only a message with the severity level *[INFO]*, the
+outcome of the Test Case is "pass".
```

Comment:
Make the Outcome section identical to the Outcome section in #1284.

Also, make sure that `[NOTICE]` etc are in the link list.